compiler-core/src/type_/expression.rs

```diff
@@ -3358,6 +3361,39 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
         Ok(())
     }
 
+    fn check_single_case(&mut self, clauses: &[UntypedClause]) {
+        if clauses.len() > 1 {
+            return;
+        }
+
+        let _ = clauses
+            .first()
+            .and_then(|pats| pats.pattern.first())
+            .map(|pat| {
+                if let Pattern::Constructor {
+                    name,
+                    arguments,
+                    location,
+                    ..
+                } = pat
+                {
+                    let args = arguments
+                        .iter()
+                        .filter_map(|call| match &call.value {
+                            Pattern::Variable { name, .. } => Some(name),
+                            Pattern::Discard { name, .. } => Some(name),
+                            _ => None,
+                        })
+                        .join(", ");
+                    let hint = EcoString::from(format!("let {name}({args}) = ..."));
```

Comment:
I don't think it's a good idea to reformat the pattern here. Some information, such as whether the constructor is qualified, will be lost. And, as louis pointed out, this only works for constructor patterns at the moment. What I did for the "let assert to case" action is to get the location of the pattern and copy the exact string value from the source. I'm not sure if you have access to the source here though

Also, you can use `eco_format!` instead of `format!` and `EcoString::from`